daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -299,11 +299,42 @@ object SExpr {
     }
   }
 
+  /** A (single) let-expression with an unhungry, saturated builtin-application as RHS */
+  final case class SELet1BuiltinArithmetic(
+      builtin: SBuiltinArithmetic,
+      args: Array[SExprAtomic],
+      body: SExpr,
+  ) extends SExpr
+      with SomeArrayEquals {
+    def execute(machine: Machine): Unit = {
+      val arity = builtin.arity
+      val actuals = new util.ArrayList[SValue](arity)
+      var i = 0
+      while (i < arity) {
+        val arg = args(i)
+        val v = arg.lookupValue(machine)
+        actuals.add(v)
+        i += 1
+      }
+      builtin.executePure(actuals) match {
+        case Right(value) =>
+          machine.pushEnv(value) //use pushEnv not env.add so instrumentation is updated
+          machine.ctrl = body
+        case Left(message) =>
+          machine.traceLog.add("ArithmeticError:" + message, machine.lastLocation)
```

Comment:
I feel slightly uneasy about abusing the tracelog for this and mixing user logs with internal logs produced by speedy. Fine for this PR but I think we should separate the two.

Reply:
Looking at the log above, it also appears that you log the same error twice. Once here once in `SEBuiltinArithmetic`.